### PR TITLE
[Compute] Update Compute client library to 23.2.0

### PR DESCRIPTION
**Pull request by hyonholee**
Add-AzureRmVmssVMDataDisk and Remove-AzureRmVmssVMDataDisk

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/181

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by hyonholee** [comment]
@praries880 Obsolete makes a build failure.  Shall I just add a warning message?

File: src/ResourceManager/Compute/Commands.Compute/AzureRM.Compute.psd1 [hunk 1]
```diff
@@ -181,7 +181,8 @@ CmdletsToExport = 'Remove-AzureRmAvailabilitySet', 'Get-AzureRmAvailabilitySet',
                'New-AzureRmGalleryImageDefinition', 'Update-AzureRmGalleryImageDefinition',
                'Get-AzureRmGalleryImageDefinition', 'Remove-AzureRmGalleryImageDefinition',
                'New-AzureRmGalleryImageVersion', 'Update-AzureRmGalleryImageVersion',
-               'Get-AzureRmGalleryImageVersion', 'Remove-AzureRmGalleryImageVersion'
+               'Get-AzureRmGalleryImageVersion', 'Remove-AzureRmGalleryImageVersion',
+               'Add-AzureRmVmssVMDataDisk', 'Remove-AzureRmVmssVMDataDisk'
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/AzureRM.Compute.psd1]: @hyonholee please make sure these cmdlets are added to `Az.Compute.psd1` as well

File: src/ResourceManager/Compute/Commands.Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs [hunk 2]
```diff
@@ -339,24 +339,36 @@ private void Run()
             {
                 if (vVirtualMachineProfile == null)
                 {
-                    vVirtualMachineProfile = new Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetVMProfile();
+                    vVirtualMachineProfile = new VirtualMachineScaleSetVMProfile();
                 }
                 vVirtualMachineProfile.StorageProfile = this.StorageProfile;
             }
 
+			if (this.EnableUltraSSD.IsPresent)
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs]: @hyonholee nit: looks like an issue with tabbing/spacing here

File: src/ResourceManager/Compute/Commands.Compute/Manual/AddAzureRmVmssVMDataDiskCommand.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Compute.Automation.Models;
+using Microsoft.Azure.Commands.Compute.Common;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Compute.Models;
+
+namespace Microsoft.Azure.Commands.Compute.Automation
+{
+    [Cmdlet("Add", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VmssVMDataDisk")]
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Manual/AddAzureRmVmssVMDataDiskCommand.cs]: @hyonholee nit: please replace `"Add"` with `VerbsCommon.Add`

File: src/ResourceManager/Compute/Commands.Compute/Manual/AddAzureRmVmssVMDataDiskCommand.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    [OutputType(typeof (PSVirtualMachineScaleSetVM))]
+    public class AddAzureRmVmssVMDataDiskCommand : ComputeClientBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Manual/AddAzureRmVmssVMDataDiskCommand.cs]: @hyonholee you can remove `ValueFromPipelineByPropertyName = true` for this parameter

File: src/ResourceManager/Compute/Commands.Compute/Manual/RemoveAzureRmVmssVMDataDiskCommand.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+// ----------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Compute.Automation.Models;
+using Microsoft.Azure.Commands.Compute.Common;
+
+namespace Microsoft.Azure.Commands.Compute.Automation
+{
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VmssVMDataDisk")]
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Manual/RemoveAzureRmVmssVMDataDiskCommand.cs]: @hyonholee nit: please replace `"Remove"` with `VerbsCommon.Remove`

File: src/ResourceManager/Compute/Commands.Compute/Manual/RemoveAzureRmVmssVMDataDiskCommand.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    [OutputType(typeof(PSVirtualMachineScaleSetVM))]
+    public class RemoveAzureRmVmssVMDataDiskCommand : Microsoft.Azure.Commands.ResourceManager.Common.AzureRMCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Manual/RemoveAzureRmVmssVMDataDiskCommand.cs]: @hyonholee same comment about removing `ValueFromPipelineByPropertyName = true` from this parameter

**Comment by maddieclayton** [comment]
@markcowl The SDK client referenced has been upgraded to Newtonsoft Json 10, so this should be ready to merge when code review is complete.

**Pull request merged**